src/cpprealm/internal/bridge/realm.hpp

```diff
@@ -47,7 +47,30 @@ namespace realm::internal::bridge {
             RecoverOrDiscard,
         };
 
-        struct sync_config;
+        struct sync_config {
+
+            struct proxy_config {
+                using port_type = std::uint_fast16_t;
+                enum class Type { HTTP, HTTPS } type;
```

Comment:
We never implemented support for HTTPS proxies, I don't think we should expose this as an option here.